### DISCUSSION TITLE
improving error reporting while processing doc metadata feasible?

### DISCUSSION THREAD

**Discussion by mi-hol** [facebook/docusaurus#9134] [General]
I got below error.  Finding the root cause was difficult, because the line causing this error is not displayed.

```
> docusaurus build

[INFO] [en] Creating an optimized production build...
Error:  Error while parsing Markdown front matter.
This can happen if you use special characters in front matter values (try using double quotes around that value).
Error:  Loading of version failed for version current
[INFO] Docusaurus version: 2.4.1
Node version: v18.16.1
Error:  Unable to build website for locale en.
Error:  Error: Can't process doc metadata for doc at path path=/home/runner/work/wiki/wiki/docs/01-Welcome-to-FOME.md in version name=current
    at processDocMetadata (/home/runner/work/wiki/wiki/node_modules/@docusaurus/plugin-content-docs/lib/docs.js:200:[15](https://github.com/FOME-Tech/wiki/actions/runs/5494974336/jobs/10014069631#step:5:16))
    at async Promise.all (index 0)
Error: Process completed with exit code 1.
```

**to reproduce** add a md comment inside doc metadata

```md
---
slug: /

[//]: # (Docusaurus 2.4.1 does NOT allow md comments inside doc metadata

---
```

Is this change feasible?

**Comment by Josh-Cena** [comment]
We don't know the line causing the error, because we are trying to parse front matter but you did not provide valid front matter; that's all we know. I'm fairly sure what you have there is not valid YAML. The front matter is NOT Markdown.

```md
---
slug: /
# You can use YAML comments
---
```

**Reply by mi-hol** [reply]
thanks, I didn't know that front matter is using YAML syntax.
I didn't see this fact and "how to add a comment" in https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#markdown-front-matter
Would it make sense to add it there or is this already mentioned somewhere else?

**Reply by Josh-Cena** [reply]
Yes: https://docusaurus.io/docs/markdown-features#front-matter

Frankly, front matter is a quite common feature in Markdown, so there are also other resources available on the Internet.

**Reply by mi-hol** [reply]
> Frankly, front matter is a quite common feature in Markdown,

Unfortunately coming from github's Markdown to Docusaurus I had no previous exposure to front matter.
For experts everything in their domain is easy but newbies require time to learn based on errors they make.
This is the time when meaningful error messages play an important role.
The 2 current error messages quoted fail to set a proper context what to check for a newbie.
To give this a positive spin, based on the first answer the Docusaurus code parsing front matter seems to assume a correct YAML structure but does not actually verify it as first action.
An error message like "Error: metadata written in front matter is not in valid YAML format." would have brought me on the right track.


Would  adding  a check along the lines "is the supplied front matter text in valid YAML format" make sense as an enhancement?

**Reply by slorber** [reply]
> Would adding a check along the lines "is the supplied front matter text in valid YAML format" make sense as an enhancement?

I'm happy to improve the error message if that helps, please submit a PR

**Comment by mi-hol** [comment]
> I'm happy to improve the error message if that helps, please submit a PR

As said I'm a newbie to Docusaurus and can only report symptoms of an issue but can't cure them :(
The image of an **expert only tool** seems correct as the response to [my doc update PR](https://github.com/facebook/docusaurus/pull/9137) clearly shows
